src/compiler/GF/Server.hs

```diff
@@ -407,9 +407,6 @@ resp404 path = Response 404 [plain,xo] $ "Not found: "++path++"\n"
 resp500 msg  = Response 500 [plain,xo] $ "Internal error: "++msg++"\n"
 resp501 msg  = Response 501 [plain,xo] $ "Not implemented: "++msg++"\n"
 
-instance Error Response where
-  noMsg = resp500 "no message"
-  strMsg = resp500
```

Comment:
Do we no longer need this instance?

Reply:
There is a slight difference between `ErrorT` and `ExceptT`, specifically in the `MonadFail` instance. When calling `fail`, `ErrorT` would [use the `strMsg` function](https://hackage.haskell.org/package/transformers-0.5.6.2/docs/src/Control.Monad.Trans.Error.html#local-6989586621679042850) above and return a `Left` value, while `ExceptT` would instead call the `fail` function of the monad above (in this case `IO`) which would throw an error locally.

So, if at any point the code here would call `fail`, the error would not be forwarded to the client properly, but the server would instead crash. In conclusion, I'm not completely sure if replacing `ErrorT` with `ExceptT` is safe.

Reply:
Thanks for digging into this @anka-213 !

So we should let ErrorT be for now, because ExceptT has actually different behaviour on failure. In order to prepare for future (if GHC stops supporting ErrorT), someone could write tests, so we can be a bit more secure in doing the change.

Reply:
There are on the other hand [examples](https://github.com/GrammaticalFramework/gf-core/blob/7dc6717b5e95122a810d3b580981fc4ce0c9744b/src/compiler/GF/Server.hs#L230) of the same issue already in this file. And I can't find any uses of `fail` in the module, so maybe it is fine. Or maybe the other code should be fixed as well.

Reply:
Hi @anka-213 ! 

> Or maybe the other code should be fixed as well.

By this, did you mean I should leave it as ErrorT, as what @inariksit said ?